specs/phase1/beacon-chain.md

```diff
@@ -886,9 +886,11 @@ def process_crosslink_for_shard(state: BeaconState,
         for attestation in transition_attestations:
             participants = get_attesting_indices(state, attestation.data, attestation.aggregation_bits)
             transition_participants = transition_participants.union(participants)
-            assert attestation.data.shard_head_root == shard_transition.shard_data_roots[
-                len(shard_transition.shard_data_roots) - 1
-            ]
+            if len(shard_transition.shard_states) > 0:
+                # Is the `shard_transition` candidate
+                last_offset_index = len(shard_transition.shard_states) - 1
+                shard_head_root = shard_transition.shard_states[last_offset_index].latest_block_root
+                assert attestation.data.shard_head_root == shard_head_root
```

Comment:
Second thought.... What do you think about moving this to `validate_attestation`?  Feels more natural and more obvious that it's an on-time check.

```python
    # Type 1: on-time attestations
    if is_on_time_attestation(state, attestation):
     # Assert correct parent block root
     # Assert correct shard number
     # Assert shard head root is consistent with latest shard state block root
     ...
```

Reply:
I feel the same, but it would break the compatibility of `process_attestation` interface (adding `shard_transitions: Seqence[ShardTransition]`) from phase0 to phase1. Especially, we currently have some phase 0 test generators that are compatible with both phase0 and phase1. If we change the interface, the test case and test format would be messier. 

That said, IMO it's worth to reconsider when phase1 spec content is complete.

Reply:
Ah yes. I forgot about about the compatibility. Reconsider at a later date sounds good to me